csharp/src/Google.Protobuf/Properties/AssemblyInfo.cs

```diff
@@ -59,14 +59,18 @@
 
 [assembly: AssemblyVersion("2.4.1.555")]
 
+#if SIGNED
 [assembly: InternalsVisibleTo("Google.Protobuf.Test, PublicKey=" +

```

Comment:
Yep, I saw the approach. it's good once you have more InternalsVisibleTo clauses. We currently have just one. Let's switch to your approach once we have a need for it.